examples/naming_package_files/my_package_name.bzl

```diff
@@ -82,3 +82,19 @@ names_from_toolchains = rule(
     },
     toolchains = ["@rules_cc//cc:toolchain_type"],
 )
+
+#
+# Using a command line build setting to name a package.
+#
+def _label_from_command_line_naming_impl(ctx):
+    values = {"label": ctx.build_setting_value}
+    return PackageVariablesInfo(values = values)
+
+#
+# Creating this build_setting defines a flag the user can set.
+#
+label_from_command_line = rule(
+    implementation = _label_from_command_line_naming_impl,
+    # Note that the default value comes from the rule instantiation.
+    build_setting = config.string(flag = True),
+)
```

Comment:
What about using bazel_skylib's [`string_flag`](https://github.com/bazelbuild/bazel-skylib/blob/master/docs/common_settings_doc.md#string_flag) rule?

My understanding is that most people should be using it rather than rolling their own rule like this.

Reply:
The string_flag provides a `BuildSettingInfo`, which is singled valued, so we would have to remap to a `PackageVariablesInfo`.    I think that would introduce a layer of complexity that might distract from the point.

Reply:
Oooh.  Right.  Totally missed that.  My bad.